src/lm_pub_quiz/integration.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+from transformers import TrainerCallback
+
+from lm_pub_quiz.data import Dataset
+from lm_pub_quiz.evaluator import Evaluator
+
+
+class PubQuizCallback(TrainerCallback):
+    """This callback handels on the fly BEAR evaluation for the huggingface Trainer class"""
+    def __init__(self, trainer, dataset_path, metrics=["accuracy"], model_type=None, save_path=None, batch_size=32, template=0):
+        self.trainer = trainer
+        self.model_type = model_type
+        self.save_path = save_path
+        self.batch_size = batch_size
+        self.template = template
+        if "accuracy" in metrics and "num_instances" not in metrics:
+            metrics.append("num_instances")
+        self.metrics = metrics
+        # include default loading of dataset from url?
```

Comment:
I think it may already be possible to initialize it given the name (currently only BEAR or BEAR-big).